src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -159,8 +159,6 @@ public String getApiUri() {
         } catch (RateLimitExceededException rle) {
             listener.getLogger().format("%n%s%n%n", rle.getMessage());
             throw new InterruptedException();
-        } catch (IOException e) {
-            // may be a user... ok to ignore
```

Comment:
The _real fix_ was to remove all `return` before `observer.observe(head, hash)` (in `GitHubSCMSource`) or ` observer.observe(name)` (in `GitHubSCMNavigator`) are invoked.

And remove `catch (IOException e)` because we can be muting network issues.